text/0000-security-advisories.md

```diff
@@ -0,0 +1,259 @@
+- Feature Name: security_advisories
+- Start Date: 2016-08-24
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Crates.io should offer an API to release security advisories for crates. Cargo
+should offer a new command for the same purpose and warn about vulnerable crate
+versions during compilation.
+
+# Motivation
+[motivation]: #motivation
+
+When compared to other ecosystems such as Python's, Rust's broader community
+prefers many single-purpose crates over larger monoliths. This situation,
+together with the strongly encouraged practice of pinning MINOR versions of
+dependencies, slows down the propagation of critical security fixes.
```

Comment:
I think it's worth separating two concerns: notifications and dealing with upgrades.

Regardless of any semver issues it's important for end-users (crate `W` author from the `W->X->Y->Z` example) to be notified of any dependencies being vulnerable. This need stands on its own and might be the biggest motivator.

Then once you're aware of the vulnerabilities there's the case of `cargo update` not working for some reason (major version bump, version pining, etc). Whatever the reason, one might want cargo or crates.io to impose some limitations in this situation, but perhaps not bother the majority of crate authors who don't pin minor versions.